properties.schema

```diff
@@ -84,6 +84,62 @@
         }
       }
     },
+    "_tooltips": {
+      "type": "object",
+      "required": false,
+      "title": "Tooltips",
+      "properties": {
+        "_isEnabled": {
+          "type": "boolean",
+          "required": true,
+          "default": false,
+          "title": "Enable tooltips",
+          "inputType": "Checkbox",
+          "validators": [],
+          "Help": "If enabled, tooltips defined for each item will be displayed."
+        },
+        "_desktopShowOnHover": {
+          "type": "boolean",
+          "required": true,
+          "default": true,
+          "title": "Show tooltips only on hover on desktop",
+          "inputType": "Checkbox",
+          "validators": [],
+          "Help": "Desktop only setting. If enabled, tooltips will only display when the cursor hovers over the relevent pin."
+        },
+        "_alwaysShowOnTouch": {
+          "type": "boolean",
+          "required": true,
+          "default": true,
+          "title": "Always display tooltips on touch devices.",
+          "inputType": "Checkbox",
+          "validators": [],
+          "Help": "Touch device only setting. If enabled, tooltips will always render on touch devices. If not enabled, tooltips will not be sown on touch devices."
+        },
+        "alignment": {
+          "type": "string",
+          "required": true,
+          "default": "top",
+          "title": "Tooltip alignment:",
+          "enum": ["left", "top", "right", "bottom"],
+          "inputType": {
+            "type": "Select",
+            "options": ["left", "top", "right", "bottom"]
+          },
+          "validators": [],
+          "help": "Choose the preferred position for the tooltip, relative to the item pin. If there is not enough room within the parent div to display the tooltip there, it will automatically be displayed at the opposite alignment."
+        },
+        "margin": {
+          "type": "number",
+          "required": false,
+          "default": 0,
```

Comment:
can you set this to a good practical default and/or could `margin` be achieved with styling only?

Reply:
I think a practical default is the way to go here. Setting a margin to all 4 sides throws the tooltip out of alignment, and setting a margin for 1 side only works as long as there is space for the tooltip to display- otherwise it will display on the opposite side and the margin will have to change too.

Users would need to set the styling up for margin-left / margin-right when using 'left' or 'right' alignment, for example. And then the equivalent for 'top' or 'bottom'. 

It's not much extra setup, but the logic is already in there to sort out margins if they are defined in components.json.

Reply:
Styling shouldn't really be configured per component instance unless it's going to vary per component. 
A `top`,`left`,`bottom` or `right` class with a default styling would also work.

Reply:
@guywillis agrees with you and says "there is probably some use case where on a per component basis we the user may elect to change the distance from the pin without editing the theme" but, as this code hasn't been tested in an AAT, we've no idea if that use case would ever be realised. 

i think i'm happy for you to leave it as it is. but it is quite irregular to have stylistic properties defined in the json on a per component instance basis.